front/lib/api/llm/tests/conversations.ts

```diff
@@ -302,23 +316,36 @@ export const runConversation = async (
       expect(outputTokens).toBeGreaterThan(0);
     }
 
-    if (expectedInResponse !== null) {
-      switch (expectedInResponse.type) {
-        case "text_contains":
-          expect(fullResponse.toLowerCase()).toContain(
-            expectedInResponse.substring.toLowerCase()
-          );
-          break;
-        case "has_tool_call":
-          const { toolName, expectedArguments } = expectedInResponse;
-          const matchingToolCall = toolCalls.find(
-            (tc) =>
-              tc.name === toolName && tc.arguments.includes(expectedArguments)
+    if (config.responseFormat) {
```

Comment:
not ideal, here we loop on all conversations and some of them don't make sense for a structured output so we don't do any expect.

maybe you have a better idea ?